### PR TITLE
[LD-6156] adds sre agents instance stack

### PR DESCRIPTION
**Pull request by MehdiZonjy**
This PR provisions a EC2 that will be used to run SRE relevant agents (currently only instana).

The stack consists of  the following resources:
- EC2
- IAM Role 
- ASG

Instana Agent will be monitoring our AWS infra resources. I'm using userdata script to initialize InstanaAgent. Instana setup script will create a systemd unit to manage the state of the agent.

Instana Key should be provided via a ssm parameter

I also fixed a tiny bug in the jprofiler script

### CODE REVIEW

**Comment by aquila-droid** [comment]
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/trainingrocket/stacks/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [d17ccbf](https://github.com/trainingrocket/stacks/commits/d17ccbf04c1944451481cb7ced14c37436693e27) [LD-6156] adds sre agents instance stack

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by aquila-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **andreshazard**
You can assign the PR to them by writing `/assign @andreshazard` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=trainingrocket%2Fstacks).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/trainingrocket/stacks/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: sre-agents/sre-agents.yaml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    Description: Amazon Linux 2 AMI
+  Product:
+    Type: String
+    Default: sre-agents
+  Business:
+    Type: String
+    Default: orange
+  Team:
+    Type: String
+    Default: ld-sre
```
Comment (naiduarvind) [inline sre-agents/sre-agents.yaml]: @MehdiZonjy - team should be just SRE. We are not bounded by country, business unit, and etc.

File: sre-agents/sre-agents.yaml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    Type: Number
+    Description: instance volume size
+    Default: 32
+  ImageId:
+    Type: AWS::EC2::Image::Id
+    Default: ami-082b5a644766e0e6f
+    Description: Amazon Linux 2 AMI
+  Product:
+    Type: String
+    Default: sre-agents
```
Comment (naiduarvind) [inline sre-agents/sre-agents.yaml]: Nothing of a huge change but would it sound better if it was `sre-utilities` or `sre-auxiliary`?

File: sre-agents/sre-agents.yaml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                - kinesis:DescribeStream
+                - kinesis:ListTagsForStream
+                - lambda:ListTags
+                - lambda:ListFunctions
+                - lambda:ListEventSourceMappings
+                - lambda:GetFunctionConfiguration
+                - s3:GetBucketTagging
+                - s3:ListAllMyBuckets
+                - s3:GetBucketLocation
+                - tag:GetResource
```
Comment (naiduarvind) [inline sre-agents/sre-agents.yaml]: We will also need to look at how the resource usage of the agent would look like with all this enabled - just something to keep in mind.

File: sre-agents/sre-agents.yaml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  VPC:
+    Type: AWS::EC2::VPC::Id
+    Description: VPC in which to deploy the instance
+  Subnets:
+    Description: The subnets where the sre instance will be created.
+    Type: List<AWS::EC2::Subnet::Id>
+  KeyPair:
+    Type: AWS::EC2::KeyPair::KeyName
+    Description: SSH Keypair for SRE Agents instance
+    Default: sre-agents
```
Comment (naiduarvind) [inline sre-agents/sre-agents.yaml]: Did we check if we have a key pair in Learndot account?

I was also thinking of writing a page in Wiki related to all resources that SRE uses.

File: sre-agents/sre-agents.yaml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+          IpProtocol: tcp
+        - CidrIp: 135.84.144.214/32 # Palo Alto
+          FromPort: 22
+          ToPort: 22
+          IpProtocol: tcp
+        - CidrIp: 216.241.29.130/32 # Santiago
+          FromPort: 22
+          ToPort: 22
+          IpProtocol: tcp
+        - CidrIp: 210.19.86.66/32 # Kuala Lumpur
```
Comment (naiduarvind) [inline sre-agents/sre-agents.yaml]: Did we miss out Sydney?

**Comment by aquila-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 15aa6e8ac0689e255cb88021d1ecaa4d0b48adc6</details>

**Comment by aquila-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/trainingrocket/stacks/pull/6#pullrequestreview-266163164" title="Approved">naiduarvind</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=trainingrocket%2Fstacks).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/trainingrocket/stacks/blob/master/OWNERS)~~ [naiduarvind]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by aquila-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/trainingrocket/stacks/pull/6#issuecomment-514719567" title="LGTM">naiduarvind</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=trainingrocket%2Fstacks).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/trainingrocket/stacks/blob/master/OWNERS)~~ [naiduarvind]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: sre-agents/sre-agents.yaml (hunk 3 above)
Comment (MehdiZonjy) [reply sre-agents/sre-agents.yaml]: Yep; in staging there is a spike of cpu usage during bootstrap; but afterwards it goes back to normal.  Prod might be different as there are more resources though

File: sre-agents/sre-agents.yaml (hunk 5 above)
Comment (MehdiZonjy) [reply sre-agents/sre-agents.yaml]: yes we did

**Comment by aquila-droid** [comment]
New changes are detected. LGTM label has been removed.

**Comment by aquila-droid** [comment]
@MehdiZonjy: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
list-buckets-to-S3 | 829e32e38f7c52bf0740905958f39cbf58d1f998 | [link](https://prow.ent.yardrat.org/view/gcs/<bucket-name>/pr-logs/pull/stacks/6/list-buckets-to-S3/1154260751829962752) | `/test list-buckets-to-S3`

[Full PR test history](https://prow.ent.yardrat.org/pr-history?org=trainingrocket&repo=stacks&pr=6)

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by aquila-droid** [comment]
LGTM label has been added.  <details>Git tree hash: 379dfa7500812597cd59f559e802825f2eb3a75e</details>

**Comment by aquila-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/trainingrocket/stacks/pull/6#issuecomment-514910581" title="LGTM">naiduarvind</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=trainingrocket%2Fstacks).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/trainingrocket/stacks/blob/master/OWNERS)~~ [naiduarvind]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by aquila-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/trainingrocket/stacks/pull/6#pullrequestreview-266417081" title="Approved">naiduarvind</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=trainingrocket%2Fstacks).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/trainingrocket/stacks/blob/master/OWNERS)~~ [naiduarvind]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**